resources/assets/js/features/user/components/Register/Form.js

```diff
@@ -49,7 +61,11 @@ class Form extends React.Component {
                     response => {
                         registerSuccess();
                         initSession(response.data.token);
-                        browserHistory.push('/dashboard');
+                        if (hasTripPending) {
+                            savePendingTrip(tripPending).then(() => completeTrip());
```

Comment:
no need hasTripPending here you can define should you create trip or not based on global state tripPending, if you have one - save, if not - don't